### PR TITLE
Adds IAM get/set for binding members and roles

### PR DESCRIPTION
**Pull request by gguuss**

### CODE REVIEW

File: iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java [hunk 1]
```diff
@@ -93,15 +93,16 @@ public static Topic createIotTopic(String projectId, String topicId) throws Exce
 
     try (TopicAdminClient topicAdminClient = TopicAdminClient.create()) {
       final Topic topic = topicAdminClient.createTopic(topicName);
-      Policy policy = topicAdminClient.getIamPolicy(topicName.toString());
+      com.google.iam.v1.Policy policy = topicAdminClient.getIamPolicy(topicName.toString());
```
Comment (lesv) [inline iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java]: Not a fan of this -- is it required?

File: iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java [hunk 2]
```diff
@@ -93,15 +93,16 @@ public static Topic createIotTopic(String projectId, String topicId) throws Exce
...
       // add role -> members binding
       Binding binding =
           Binding.newBuilder()
               .addMembers("serviceAccount:cloud-iot@system.gserviceaccount.com")
               .setRole(Role.owner().toString())
               .build();
       // create updated policy
-      Policy updatedPolicy = Policy.newBuilder(policy).addBindings(binding).build();
+      com.google.iam.v1.Policy updatedPolicy =
```
Comment (lesv) [inline iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java]: also, not a fan.

File: iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java [hunk 3]
```diff
@@ -578,6 +579,114 @@ public static void setDeviceConfiguration(
...
+    JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+    HttpRequestInitializer init = new RetryHttpInitializerWrapper(credential);
+    final CloudIot service = new CloudIot.Builder(
+        GoogleNetHttpTransport.newTrustedTransport(),jsonFactory, init)
+        .setApplicationName(APP_NAME).build();
+
+    final String registryPath = String.format("projects/%s/locations/%s/registries/%s",
+        projectId, cloudRegion, registryName);
+
+    com.google.api.services.cloudiot.v1.model.Policy policy =
```
Comment (lesv) [inline iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java]: ugly

File: iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java (hunk 2 above)
Comment (gguuss) [reply iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java]: This is the code that does not appear in documentation, it's necessary for testing the API and is a convenient helper if someone is using this to automate device provisioning.

**Comment by lesv** [comment]
@gguuss What do i need to enable?

```
 Change detected. Running tests. 
 
[ERROR] Tests run: 17, Failures: 0, Errors: 7, Skipped: 0, Time elapsed: 570.949 s <<< FAILURE! - in com.example.cloud.iot.examples.ManagerIT
[ERROR] testHttpDeviceEvent(com.example.cloud.iot.examples.ManagerIT)  Time elapsed: 3.196 s  <<< ERROR!
com.google.api.client.http.HttpResponseException: 
403 Forbidden
{
  "error": {
    "code": 403,
    "message": "User not authorized to perform this action.",
    "errors": [
      {
        "message": "User not authorized to perform this action.",
        "domain": "global",
        "reason": "forbidden"
      }
    ],
    "status": "PERMISSION_DENIED"
  }
}

	at com.example.cloud.iot.examples.ManagerIT.testHttpDeviceEvent(ManagerIT.java:321)

[ERROR] testHttpDeviceState(com.example.cloud.iot.examples.ManagerIT)  Time elapsed: 0.591 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-1517852041).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-1517852041).

[ERROR] testGetIam(com.example.cloud.iot.examples.ManagerIT)  Time elapsed: 1.58 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-1517852041).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-1517852041).

[ERROR] testCreateDeleteRsaDevice(com.example.cloud.iot.examples.ManagerIT)  Time elapsed: 1.504 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-1517852041).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-1517852041).

[ERROR] testCreateDeleteUnauthDevice(com.example.cloud.iot.examples.ManagerIT)  Time elapsed: 0.694 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-1517852041).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-1517852041).

[ERROR] testSetIam(com.example.cloud.iot.examples.ManagerIT)  Time elapsed: 0.74 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-1517852041).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-1517852041).

[ERROR] testPatchRsa(com.example.cloud.iot.examples.ManagerIT)  Time elapsed: 0.852 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-1517852041).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-1517852041).

[ERROR] Errors: 
[ERROR]   ManagerIT.testCreateDeleteRsaDevice ? AlreadyExists io.grpc.StatusRuntimeExcep...
[ERROR]   ManagerIT.testCreateDeleteUnauthDevice ? AlreadyExists io.grpc.StatusRuntimeEx...
[ERROR]   ManagerIT.testGetIam ? AlreadyExists io.grpc.StatusRuntimeException: ALREADY_E...
[ERROR]   ManagerIT.testHttpDeviceEvent:321 ? HttpResponse 403 Forbidden
{
  "error": {
...
[ERROR]   ManagerIT.testHttpDeviceState ? AlreadyExists io.grpc.StatusRuntimeException: ...
[ERROR]   ManagerIT.testPatchRsa ? AlreadyExists io.grpc.StatusRuntimeException: ALREADY...
[ERROR]   ManagerIT.testSetIam ? AlreadyExists io.grpc.StatusRuntimeException: ALREADY_E...
[ERROR] Tests run: 17, Failures: 0, Errors: 7, Skipped: 0
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:2.20.1:verify (default) on project cloudiot-manager-demo: There are test failures.
[ERROR] 
[ERROR] Please refer to /tmpfs/src/github/java-docs-samples/iot/api-client/manager/target/failsafe-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException


[ID: 6350789] Build finished after 673 secs, exit value: 1


Warning: Permanently added 'localhost' (ECDSA) to the list of known hosts.
Build script failed with exit code: 1
```

**Pull request merged**